raphtory/src/algorithms/motifs/three_node_temporal_motifs.rs

```diff
@@ -127,54 +113,82 @@ pub fn star_motif_count<G: GraphViewOps>(
             }
         })
         .collect::<Vec<StarEvent>>();
-    exploded_edges.sort_by_key(|e| (e.time, e.dir));
-    let mut star_count = init_star_count(neigh_map.len());
-    star_count.execute(&exploded_edges, delta);
-    star_count.return_counts()
+
+    deltas
+        .into_iter()
+        .map(|delta| {
+            let mut star_count = init_star_count(evv.degree());
+            star_count.execute(&events, delta);
+            star_count.return_counts()
+        })
+        .collect::<Vec<[usize; 24]>>()
```

Comment:
You are redoing the count separately for all the deltas here and in other places but the results are nested as a motif for small delta is still there for large delta, right? Should be possible to use that to speed up the computation a lot as you can save many comparisons.